pkg/watchdog/watchdog.go

```diff
@@ -38,19 +38,30 @@ func WatchdogFileDirectory(baseDir string) string {
 	return filepath.Join(baseDir, "watchdog-files")
 }
 
-func WatchdogFileFromNamespaceName(baseDir string, namespace string, name string) string {
-	watchdogFile := namespace + "_" + name
+func WatchdogFileFromNamespaceNameUID(baseDir string, namespace string, name string, uid string) string {
+	watchdogFile := namespace + "_" + name + "_" + uid
 	return filepath.Join(baseDir, "watchdog-files", watchdogFile)
 }
 
 func WatchdogFileRemove(baseDir string, vmi *v1.VirtualMachineInstance) error {
 	namespace := precond.MustNotBeEmpty(vmi.GetObjectMeta().GetNamespace())
 	domain := precond.MustNotBeEmpty(vmi.GetObjectMeta().GetName())
 
-	file := WatchdogFileFromNamespaceName(baseDir, namespace, domain)
+	path := WatchdogFileFromNamespaceNameUID(baseDir, namespace, domain, string(vmi.UID))
+	files, err := filepath.Glob(path + "*")
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to get files from %s", path)
+		return err
+	}
 
-	log.Log.V(3).Infof("Remove watchdog file %s", file)
-	return diskutils.RemoveFile(file)
+	for _, file := range files {
+		log.Log.V(3).Object(vmi).Infof("Remove watchdog file %s", file)
+		err := diskutils.RemoveFile(file)
```

Comment:
here too, what leads to multiple files now that we're putting the UID in the file name? I wouldn't expect this to be necessary

Reply:
The same situation.